File: easy/3838_weighted_word_mapping/solution.py
from typing import List


class Solution:
    def mapWordWeights(self, words: List[str], weights: List[int]) -> str:
        letters = [0] * len(words)

        for i, word in enumerate(words):
            for c in word:
                letters[i] += weights[ord(c) - 97]

            letters[i] = chr(122 - (letters[i] % 26))

        return "".join(letters)

File: easy/3838_weighted_word_mapping/solution.cpp
#include <string>
#include <vector>

class Solution {
public:
  std::string mapWordWeights(std::vector<std::string> &words,
                             std::vector<int> &weights) {
    std::string letters;
    letters.reserve(words.size());

    for (const std::string &word : words) {
      int val = 0;
      for (const char &c : word)
        val += weights[c - 97];

      letters.push_back(122 - (val % 26));
    }

    return letters;
  }
};
